server/src/main/java/org/elasticsearch/transport/RemoteClusterConnection.java

```diff
@@ -357,201 +358,161 @@ public String getProxyAddress() {
-         * trigger a connect round if there is no listener queued up.
-         */
-        void forceConnect() {
-            connect(null, true);
-        }
-
-        private void connect(ActionListener<Void> connectListener, boolean forceRun) {
             final boolean runConnect;
-            final Collection<ActionListener<Void>> toNotify;
-            final ActionListener<Void> listener = connectListener == null ? null :
+            final ActionListener<Void> listener =
                 ContextPreservingActionListener.wrapPreservingContext(connectListener, threadPool.getThreadContext());
-            synchronized (queue) {
-                if (listener != null && queue.offer(listener) == false) {
-                    listener.onFailure(new RejectedExecutionException("connect queue is full"));
+            synchronized (mutex) {
+                if (closed.get()) {
+                    if (connectListener != null) {
+                        connectListener.onFailure(new AlreadyClosedException("connect handler is already closed"));
+                    }
                     return;
                 }
-                if (forceRun == false && queue.isEmpty()) {
+                if (queue.offer(listener) == false) {
+                    listener.onFailure(new RejectedExecutionException("connect queue is full"));
                     return;
                 }
-                runConnect = running.tryAcquire();
-                if (runConnect) {
-                    toNotify = new ArrayList<>();
-                    queue.drainTo(toNotify);
-                    if (closed.get()) {
-                        running.release();
-                        ActionListener.onFailure(toNotify, new AlreadyClosedException("connect handler is already closed"));
-                        return;
-                    }
-                } else {
-                    toNotify = Collections.emptyList();
-                }
+                runConnect = queue.size() == 1;
             }
             if (runConnect) {
-                forkConnect(toNotify);
-            }
-        }
-
-        private void forkConnect(final Collection<ActionListener<Void>> toNotify) {
-            ExecutorService executor = threadPool.executor(ThreadPool.Names.MANAGEMENT);
-            executor.submit(new AbstractRunnable() {
-                @Override
-                public void onFailure(Exception e) {
-                    synchronized (queue) {
-                        running.release();
-                    }
-                    try {
+                ExecutorService executor = threadPool.executor(ThreadPool.Names.MANAGEMENT);
+                executor.submit(new AbstractRunnable() {
+                    @Override
+                    public void onFailure(Exception e) {
+                        final Collection<ActionListener<Void>> toNotify = new ArrayList<>();
+                        synchronized (mutex) {
```

Comment:
Do we even need these `synchronized` blocks? `drainTo` is atomic anyway so the mechanics of `runConnect = queue.size() == 1;` shouldn't be affected? (imo not, we're using `runConnect` outside the sychronized block anyway)
The same even holds true for `close` it seems, since we're not guarding what's inside ` if (runConnect) {` against a concurrent close it seems to me all but the synchronized block above https://github.com/elastic/elasticsearch/pull/44825/files#diff-d390b91a2cae82a4f90f1386f8428695R373 are technically redundant?